### ISSUE TITLE
[Enhancement]: Add `manualCreation` argument to `aws_codebuild_webhook` resource

### ISSUE THREAD

**Issue by mxxk** [hashicorp/terraform-provider-aws#40078]
### Description

The `aws_codebuild_webhook` resource presently supports most arguments corresponding to the CodeBuild API [Webhook](https://docs.aws.amazon.com/codebuild/latest/APIReference/API_Webhook.html) data type, but `manualCreation` is not one of them:

![image](https://github.com/user-attachments/assets/3f76136f-0b63-496a-9199-14ef210b2c34)

Please add an argument (presumably called `manual_creation`) to the `aws_codebuild_webhook` resource.

### Affected Resource(s) and/or Data Source(s)

* aws_codebuild_webhook

### Potential Terraform Configuration

```terraform
resource "aws_codebuild_webhook" "github_actions" {
  project_name    = aws_codebuild_project.github_actions.name
  build_type      = "BUILD"
  manual_creation = true
}
```

### References

https://docs.aws.amazon.com/codebuild/latest/APIReference/API_Webhook.html

### Would you like to implement a fix?

Yes

**Comment by avidspartan1** [comment]
This issue is now much more relevant since AWS made a breaking change to the way Codebuild webhook creation works: https://aws.amazon.com/about-aws/whats-new/2025/02/aws-codebuild-managed-webhooks-github-enterprise/. The `manualCreation` parameter now defaults to "false", meaning Codebuild will always attempt to create the Github webhook unless `manualCreation=true` is specifically passed.

If you were previously using a Codebuild source credential with essentially "read only" permissions, you must now elevate it to admin-level (hook management) permissions in order to continue using the provider.

**Comment by kieran-lowe** [comment]
Yeah this has become a real issue for us. We use Terraform to manage this for quite some time and this change has broken it.
